packages/epinio/generated-changes/patch/charts/epinio-ui/values.yaml.patch

```diff
@@ -0,0 +1,12 @@
+--- charts-original/charts/epinio-ui/values.yaml
++++ charts/charts/epinio-ui/values.yaml
+@@ -1,5 +1,8 @@
+ epinioUI:
+-  image: ghcr.io/epinio/epinio-ui:v0.6.0-0.0.1
++  image:
++    registry: ghcr.io
++    repository: epinio/epinio-ui
++    tag: v0.6.0-0.0.1
```

Comment:
There's a few things to address here (Applies to all images in this chart):

1. This image need to be mirrored - To do that you will need to submit a ticket to EIO and then add the image to the image-list file in [rancher/image-mirror](https://github.com/rancher/image-mirror) (You'll find more information on what you need to do exactly in the repo's readme).

2. This needs to follow our pattern for private registry support - Take the image in [this](https://github.com/rancher/charts/blob/1a0e0f04bc4b2e09986d381a6b112e686779b008/charts/rancher-gatekeeper/100.0.1%2Bup3.6.0/values.yaml#L26-L29) chart as an example. You first need to add [this](https://github.com/rancher/charts/blob/1a0e0f04bc4b2e09986d381a6b112e686779b008/charts/rancher-gatekeeper/100.0.1%2Bup3.6.0/values.yaml#L86-L88) to your values.yaml, then you would template your image name like [this](https://github.com/rancher/charts/blob/1a0e0f04bc4b2e09986d381a6b112e686779b008/charts/rancher-gatekeeper/100.0.1%2Bup3.6.0/templates/gatekeeper-audit-deployment.yaml#L68).

3. Due to how images are exported for airgap setups, you will need to consolidate the registry and repository keys in order for the images in this chart to be picked up by the exported during the Rancher build process.
```
# After it's been mirrored it should end up being something like this
    repository: rancher/mirrored-epinio-epinio-ui
    tag: v0.6.0-0.0.1
```

Reply:
Questions and notes

  1. Where is EIO and its ticket submission page ?

     Searching the `rancher/` org I only find [`rancher/eio-charts`](https://github.com/rancher/eio-charts) which to me does not look like what I need.

     Actually, would simply making a PR against [`rancher/image-mirror`](https://github.com/rancher/image-mirror) adding the image(s) serve the same purpose as the ticket ?

  2. AFAICT I have already done this for the chart, main and children.

     Relevant places:

       1. [main images](https://github.com/epinio/helm-charts/blob/52cc3f4a8e54629221f0f078df06243aac220bea/chart/epinio/values.yaml#L11-L27)
       2. [system default registry](https://github.com/epinio/charts/blob/6206045df1905e21670d4cdac64800d3185ad7ee/packages/epinio/generated-changes/patch/values.yaml.patch#L18-L23)
       3. Similarly for the epinio-ui, kubed, and minio charts, post-patching

            a. [ui images](https://github.com/epinio/charts/blob/epinio-1248-add-to-marketplace/packages/epinio/generated-changes/patch/charts/epinio-ui/values.yaml.patch)
	    b. [minio images](https://github.com/epinio/charts/blob/epinio-1248-add-to-marketplace/packages/epinio/generated-changes/patch/charts/minio/values.yaml.patch)

     Associated template helpers.

       1. Patch: [main helper](https://github.com/epinio/charts/blob/6206045df1905e21670d4cdac64800d3185ad7ee/packages/epinio/generated-changes/patch/templates/_helpers.tpl.patch#L6-L12)
       	  Upstream: [main helper](https://github.com/epinio/helm-charts/blob/52cc3f4a8e54629221f0f078df06243aac220bea/chart/epinio/templates/_helpers.tpl#L1-L8)
       2. Patch: [ui helper](https://github.com/epinio/charts/blob/6206045df1905e21670d4cdac64800d3185ad7ee/packages/epinio/generated-changes/patch/charts/epinio-ui/templates/_helpers.tpl.patch#L27-L36)
       3. Patch: [kubed helper](https://github.com/epinio/charts/blob/6206045df1905e21670d4cdac64800d3185ad7ee/packages/epinio/generated-changes/patch/charts/kubed/templates/_helpers.tpl.patch#L27-L36)
       4. Patch: [minio helper](https://github.com/epinio/charts/blob/6206045df1905e21670d4cdac64800d3185ad7ee/packages/epinio/generated-changes/patch/charts/minio/templates/_helpers.tpl.patch#L27-L34)

     Image References. Note: I did not bother to list here all the places which have image refs.

       1. Upstream: [main image ref](https://github.com/epinio/helm-charts/blob/52cc3f4a8e54629221f0f078df06243aac220bea/chart/epinio/templates/server.yaml#L291)
       2. Patch: [ui image ref](https://github.com/epinio/charts/blob/6206045df1905e21670d4cdac64800d3185ad7ee/packages/epinio/generated-changes/patch/charts/epinio-ui/templates/server.yaml.patch#L8)
       3. Patch: [kubed image ref](https://github.com/epinio/charts/blob/6206045df1905e21670d4cdac64800d3185ad7ee/packages/epinio/generated-changes/patch/charts/kubed/templates/deployment.yaml.patch#L8)
       4. Patch: [minio image ref](https://github.com/epinio/charts/blob/6206045df1905e21670d4cdac64800d3185ad7ee/packages/epinio/generated-changes/patch/charts/minio/templates/deployment.yaml.patch#L8)

  3. My reading of item 3 is that I am patching the mirrored images in as part of this PR,  and upstream is left untouched.
     Is that reading correct ?

Reply:
@andreas-kupries 
1. First you create a PR to rancher/image-mirror that should have the name of the repo you expect (there is a pattern, see the image-list file). Then as stated here, you open a PR to rancherlabs/eio with the necessary information (I will send you a reference PR, I just can't post it here).

2. Yes, as far as I can tell you've addressed this already. This would already be fairly easy to test using `helm template`.

3. Yes, that's correct. However, another point I wanted to bring up is that the more things you patch here on the Rancher side, the more difficult it could make future maintenance because at the time you want to update the chart to latest upstream and update the URL in package.yaml, you will likely run into several patch/merge conflicts that will need to be addressed manually. If you need all those changes patched here (including the images), that's totally fine, but anything that you think can live upstream instead would definitely alleviate the burden.

Reply:
> Then as stated [here](https://github.com/rancher/image-mirror/#adding-new-images), you ...

Note: At the referenced section it only mentions EIO with no reference to rancherlabs. I guess it is common knowledge to you that EIO implies rancherlabs. Unfortunately not for us here.

On the patching front ... Yes, I agree. As much as possible should be in upstream. Will work on that today. I am limited to epinio and epinio-ui for that though, because kubed and minio are charts we have no control over.

Reply:
State today ...

   - Moved some of the PR integration patches to our upstream charts. Only for charts under our control (epinio, epinio-ui). Only the remainder of the image handling.
   - On the converse: All patches to the foreign charts (kubed, minio) are kept local. All the patches inserting the tolerations and node selectors are kept local.

I now have to drive our (epinio) side towards making a proper release of epinio and epinio-ui, charts, and images with the features we want, so that I can

   - direct the integration towards a proper helm chart instead of a dev commit
   - make the PR for mirroring the images of that release
   - finalize the integration patches, i.e. use the images in the mirror.

And now switching to the CAP/BIT distraction.

Reply:
Created EIO request - https://github.com/rancherlabs/eio/issues/997